ksqldb-engine/src/main/java/io/confluent/ksql/serde/SerdeFeaturesFactory.java

```diff
@@ -112,37 +116,59 @@ public static SerdeFeatures buildValueFeatures(
    * classes to ensure that anytime a key is changed we properly set the key format.
    *
    * @param keyFormat un-sanitized format
-   * @param numKeyColumns number of key columns for this stream/table
+   * @param newKeyColumnSqlTypes sql types of key columns for this stream/table
    * @param allowKeyFormatChangeToSupportMultipleKeys safeguard to prevent changing key formats in
    *                                                  unexpected ways. if false, no format change
    *                                                  will take place
    * @return the new key format to use
    */
   public static KeyFormat sanitizeKeyFormat(
       final KeyFormat keyFormat,
-      final int numKeyColumns,
+      final List<SqlType> newKeyColumnSqlTypes,
       final boolean allowKeyFormatChangeToSupportMultipleKeys
```

Comment:
Let's rename this to `allowKeyFormatChangeToSupportNewKeySchema` now that this flag is used to guard key format changes for key column types as well.